src/components/Featured/FeaturedUserCard.tsx

```diff
@@ -0,0 +1,186 @@
+import { useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { FeaturedUserCardFollowFragment$key } from '~/generated/FeaturedUserCardFollowFragment.graphql';
+import { FeaturedUserCardFragment$key } from '~/generated/FeaturedUserCardFragment.graphql';
+import { useLoggedInUserId } from '~/hooks/useLoggedInUserId';
+import { pluralize } from '~/utils/string';
+
+import Badge from '../Badge/Badge';
+import breakpoints from '../core/breakpoints';
+import colors from '../core/colors';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseM, TitleM } from '../core/Text/Text';
+import FollowButton from '../Follow/FollowButton';
+
+type Props = {
+  userRef: FeaturedUserCardFragment$key;
+  queryRef: FeaturedUserCardFollowFragment$key;
+};
+
+export default function FeaturedUserCard({ userRef, queryRef }: Props) {
+  const user = useFragment(
+    graphql`
+      fragment FeaturedUserCardFragment on GalleryUser {
+        id
+        username
+        badges {
+          name
+          imageURL
+          ...BadgeFragment
+        }
+        galleries {
+          collections {
+            id
+          }
+          tokenPreviews {
+            small
+          }
+          hidden
+        }
+        ...FollowButtonUserFragment
+      }
+    `,
+    userRef
+  );
+
+  const query = useFragment(
+    graphql`
+      fragment FeaturedUserCardFollowFragment on Query {
+        ...FollowButtonQueryFragment
+        ...useLoggedInUserIdFragment
+      }
+    `,
+    queryRef
+  );
+
+  const loggedInUserId = useLoggedInUserId(query);
+  const isOwnProfile = loggedInUserId && loggedInUserId === user?.id;
+
+  const userGalleries = useMemo(() => {
+    return user?.galleries ?? [];
+  }, [user?.galleries]);
+
+  const tokenPreviews = useMemo(() => {
+    const gallery = userGalleries.find((gallery) => !gallery?.hidden);
+
+    return gallery?.tokenPreviews?.slice(0, 2) ?? [];
+  }, [userGalleries]);
+
+  const collectionCount = useMemo(() => {
+    return userGalleries.reduce((sum, gallery) => {
+      if (gallery === null || gallery.hidden || !gallery.collections) {
+        return sum;
+      }
+
+      return sum + gallery.collections.length;
+    }, 0);
+  }, [userGalleries]);
+
+  const userBadges = useMemo(() => {
+    const badges = [];
+
+    for (const badge of user?.badges ?? []) {
+      if (badge?.imageURL) {
+        badges.push(badge);
+      }
+    }
+
+    return badges;
+  }, [user?.badges]);
+
+  return (
+    <StyledFeaturedUserCard href={`/${user.username}`}>
```

Comment:
hmm true - do you know a good way to check against a null result from a useFragment (so that we can guarantee a non null user), without conditionally calling the hooks in the file?